pkg/querier/queryrange/query_range_test.go

```diff
@@ -325,6 +387,228 @@ func TestMergeAPIResponses(t *testing.T) {
+			input: []Response{
+				mustParse(t, `{"status":"success","data":{"resultType":"matrix","result":[{"metric":{"a":"b","c":"d"},"values":[[1,"1"],[2,"2"]]}],"stats":{"samples":{"totalQueryableSamples":10,"totalQueryableSamplesPerStep":[[1,5],[2,5]]}}}}`),
+				mustParse(t, `{"status":"success","data":{"resultType":"matrix","result":[{"metric":{"a":"b","c":"d"},"values":[[2,"2"],[3,"3"]]}],"stats":{"samples":{"totalQueryableSamples":20,"totalQueryableSamplesPerStep":[[2,5],[3,15]]}}}}`),
+			},
+			expected: &PrometheusResponse{
+				Status: StatusSuccess,
+				Data: PrometheusData{
+					ResultType: matrix,
+					Result: []SampleStream{
+						{
+							Labels: []cortexpb.LabelAdapter{{Name: "a", Value: "b"}, {Name: "c", Value: "d"}},
+							Samples: []cortexpb.Sample{
+								{Value: 1, TimestampMs: 1000},
+								{Value: 2, TimestampMs: 2000},
+								{Value: 3, TimestampMs: 3000},
+							},
+						},
+					},
+					Stats: &PrometheusResponseStats{Samples: &PrometheusResponseSamplesStats{
+						TotalQueryableSamples: 25,
+						TotalQueryableSamplesPerStep: []*PrometheusResponseQueryableSamplesStatsPerStep{
+							{Value: 5, TimestampMs: 1000},
+							{Value: 5, TimestampMs: 2000},
+							{Value: 15, TimestampMs: 3000},
+						},
+					}},
+				},
+			},
+		},
+		{
+			name: "[stats] Merging of samples where there is multiple partial overlaps.",
+			input: []Response{
+				mustParse(t, `{"status":"success","data":{"resultType":"matrix","result":[{"metric":{"a":"b","c":"d"},"values":[[1,"1"],[2,"2"],[3,"3"]]}],"stats":{"samples":{"totalQueryableSamples":6,"totalQueryableSamplesPerStep":[[1,1],[2,2],[3,3]]}}}}`),
+				mustParse(t, `{"status":"success","data":{"resultType":"matrix","result":[{"metric":{"c":"d","a":"b"},"values":[[2,"2"],[3,"3"],[4,"4"],[5,"5"]]}],"stats":{"samples":{"totalQueryableSamples":20,"totalQueryableSamplesPerStep":[[2,2],[3,3],[4,4],[5,5]]}}}}`),
+			},
+			expected: &PrometheusResponse{
+				Status: StatusSuccess,
+				Data: PrometheusData{
+					ResultType: matrix,
+					Result: []SampleStream{
+						{
+							Labels: []cortexpb.LabelAdapter{{Name: "a", Value: "b"}, {Name: "c", Value: "d"}},
+							Samples: []cortexpb.Sample{
+								{Value: 1, TimestampMs: 1000},
+								{Value: 2, TimestampMs: 2000},
+								{Value: 3, TimestampMs: 3000},
+								{Value: 4, TimestampMs: 4000},
+								{Value: 5, TimestampMs: 5000},
+							},
+						},
+					},
+					Stats: &PrometheusResponseStats{Samples: &PrometheusResponseSamplesStats{
+						TotalQueryableSamples: 15,
+						TotalQueryableSamplesPerStep: []*PrometheusResponseQueryableSamplesStatsPerStep{
+							{Value: 1, TimestampMs: 1000},
+							{Value: 2, TimestampMs: 2000},
+							{Value: 3, TimestampMs: 3000},
+							{Value: 4, TimestampMs: 4000},
+							{Value: 5, TimestampMs: 5000},
+						},
+					}},
+				},
+			},
+		},
+		{
+			name: "[stats] Merging of samples where there is complete overlap.",
```

Comment:
I think it will be good idea to add some test cases to show the merge function works with more than 2 series.

Reply:
done